### ISSUE TITLE
Error when parsing yaml file

### ISSUE THREAD

**Issue by krewi1** [pulumi/pulumi-kubernetes#985]
### Problem description

There is error when parsing yaml file with k8s.yaml.ConfigFile. The example yaml is:
```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: azurerunner-operator
```
Whole load yaml stuff looks like this
```typescript
export const serviceAccount = new k8s.yaml.ConfigFile("serviceAccount", {
    file: `${__dirname}/backend/service_account.yaml`
}, {
    parent: aks
});
```

### Errors & Logs

error: Running program '/current' failed with an unhandled exception:
    TypeError: Cannot read property 'map' of undefined
        at config.objs.then.configObjs (/current/node_modules/@pulumi/kubernetes/yaml/yaml.js:182:14)

### Affected product version(s)
v1.10.1

### Reproducing the issue

Sorry, not possible

**Comment by krewi1** [comment]
The error happends when we specify parent property for ComponentResourceOptions

**Comment by lukehoban** [comment]
I believe a fix for this is being worked on in  https://github.com/pulumi/pulumi-kubernetes/pull/987.

**Comment by lblackstone** [comment]
This should be fixed in v1.5.3, which includes #987
